src/forms/renderers/OAuth/index.tsx

```diff
@@ -95,146 +78,50 @@ const OAuthproviderRenderer = ({
-    //      check them down below
-    const requiredFields = useMemo(
-        () => (options ? options[Options.oauthFields] : []),
-        [options]
-    );
-
+    // Reset the configuration either to injected values or
+    //  to special default values that we can check for
     const setConfigToDefault = () => {
         const defaults = getDefaultValue(
             schema.properties,
             discriminatorProperty
         );
         handleChange(onChangePath, {
             ...defaults,
-            ...fakeDefaults,
+            ...INJECTED_VALUES,
         });
     };
 
-    const [hasAllRequiredProps, setHasAllRequiredProps] = useState(false);
-    useEffect(() => {
-        const dataKeys = Object.keys(data ?? {});
-        const nestedKeys = Object.keys(data?.[onChangePath] ?? {});
-
-        const response = every(requiredFields, (field: string) => {
-            // We know we should always have at least these two
-            if (field === CLIENT_ID || field === CLIENT_SECRET) {
-                return true;
-            }
-
-            // When inside a oneOf/anOf/etc. we need to check we have the discriminator
-            if (discriminatorProperty && field === discriminatorProperty) {
-                return true;
-            }
-
-            // Check if the field is in the main data
-            if (includes(dataKeys, field) && hasLength(data?.[field])) {
-                return true;
-            }
-
-            // As a last effort check if the field is inside of nested data
-            if (
-                includes(nestedKeys, field) &&
-                hasLength(data?.[onChangePath]?.[field])
-            ) {
-                return true;
-            }
-
-            // Didn't find anything - return false
-            return false;
-        });
-
-        setHasAllRequiredProps(response);
-    }, [data, discriminatorProperty, onChangePath, requiredFields]);
-
-    // Pull out the provider so we can render the button
-    const providerVal = options ? options[Options.oauthProvider] : NO_PROVIDER;
-    const provider = useMemo(() => startCase(providerVal), [providerVal]);
-
-    // handler for the useOauth stuff
-    const onError = (error: any) => {
-        if (error === 'access_denied') {
-            setErrorMessage(
-                intl.formatMessage(
-                    {
-                        id: 'oauth.authentication.denied',
-                    },
-                    { provider }
-                )
-            );
-        } else {
-            setErrorMessage(error);
-        }
-    };
-
-    // handler for the useOauth stuff
-    const onSuccess = async (payload: any) => {
-        const tokenResponse = await accessToken(
-            payload.state,
-            payload.code,
-            endpointConfigData
-        );
+    const { hasAllRequiredProps, showAuthenticated } = useAllRequiredPropCheck(
+        data,
+        options,
+        onChangePath,
+        discriminatorProperty
+    );
```

Comment:
Moved the logic over to a hook to keep things a bit cleaned inside this huge file.